src/App.vue

```
@@ -0,0 +1,251 @@
+<script setup>
+import { ref } from 'vue';
+import Header from './components/Header.vue';
+import LoginForm from './components/LoginForm.vue';
+import RegistrationForm from './components/RegistrationForm.vue';
+import PostList from './components/PostList.vue';
+import Sidebar from './components/Sidebar.vue';
+import Loader from './components/Loader.vue';
+import { store } from './store';
+import {
+  getUser,
+  createUser,
+  getPosts,
+  createPost,
+  updatePost,
+  deletePost,
+  getComments,
+  createComment,
+  deleteComment,
+} from './services/api';
+
+const needsRegistration = ref(false);
+const attemptedEmail = ref('');
+const sidebarMode = ref('');
+
+const handleLogin = async (email) => {
+  try {
+    const users = await getUser(email);
+    if (users.length > 0) {
+      store.user = users[0];
+      await loadPosts();
+    } else {
+      needsRegistration.value = true;
+      attemptedEmail.value = email;
+    }
+  } catch (error) {}
+};
+
+const handleRegister = async ({ email, name }) => {
+  try {
+    const newUser = await createUser({ email, name });
+    store.user = newUser;
+    needsRegistration.value = false;
+    await loadPosts();
+  } catch (error) {}
```

Similar to the login handler, errors during user registration are silently ignored. If the `createUser` call fails, the user will be stuck on the registration form without any indication of what went wrong. You should display an error message.